libr/anal/p/anal_sh.c

```diff
@@ -807,15 +1056,25 @@ static int sh_set_reg_profile(RAnal* anal){
 		"gpr	pr	.32	68	0\n"
 		"gpr	sr	.32	72	0\n"
 		"gpr	gbr	.32	76	0\n"
-		"gpr	mach	.32	80	0\n"
-		"gpr	macl	.32	84	0\n";
+		"gpr	vbr	.32	80	0\n"
+		"gpr	mach	.32	84	0\n"
+		"gpr	macl	.32	88	0\n"
+		"gpr	tmp	.32	88	0\n";
 	return r_reg_set_profile_string(anal->reg, p);
 }
 
 static int archinfo(RAnal *anal, int q) {
 	return 2; /* :) */
 }
 
+static int esil_sh_init(RAnalEsil *esil) {
```

Comment:
initializing the program counter should be task of rcore, not the esil constructor. so i would kill this function too

Reply:
I removed function, but i'm just starting to test that. There still might be dozens of bugs